added routing psr7 middleware and unittests

dropdevcoding opened the pull request:
Added simple PSR7 compatible middleware for routing

dropdevcoding committed:
added routing psr7 middleware and unittests

shochdoerfer commented on src/bitExpert/Pathfinder/Middleware/RoutingMiddleware.php:
```diff
@@ -0,0 +1,64 @@
+     *
+     * @param $attributeName
+     */
+    public function setAttributeName($attributeName)
```
Why not passing the attribute name via the constructor when creating the RoutingMiddleware instance? Maybe we should provide a getter method to be able to programmatically retrieve the current name. Otherwise you would need to hardcode the name somewhere in your codebase, maybe even in 2 different locations.

dropdevcoding replied:
Good point, so I think we could pass it as a constructor argument which is RoutingResult::class per   default.

`public function __constructor(Router $router, $attributeName = RoutingResult::class)`

dropdevcoding committed:
Added constructor arg for request attribute name where routing result will be stored in
Removed setter for request attribute name
Added getter for request attribute name to provide the possibility to use the attribute for other purposes (e.g. chaining middlewares)

dropdevcoding committed:
added missing comment for param

dropdevcoding committed:
corrected usage of renamed attribute property
removed obsolete use for RoutingResult

dropdevcoding committed:
corrected unitttest for middleware since request attribute configuration has changed

dropdevcoding committed:
composer update
